flake.nix

```diff
@@ -56,88 +55,125 @@
+              sha256 = "sha256-DjwlS8rXE7srs7A8+tHqXyUsFGtucYSeq6X0T/pVOc8=";
+            };
+
+            NIX_CFLAGS_COMPILE = toString ["-Wno-error=sign-conversion"];
+          });
+        };
       };
-      hyprland-debug = hyprland.override {debug = true;};
-      hyprland-hidpi = hyprland.override {hidpiXWayland = true;};
-      hyprland-nvidia = hyprland.override {nvidiaPatches = true;};
-      hyprland-no-hidpi = builtins.trace "hyprland-no-hidpi was removed. Please use the default package." hyprland;
-
-      udis86 = prev.callPackage ./nix/udis86.nix {};
-
-      waybar-hyprland = prev.waybar.overrideAttrs (oldAttrs: {
-        postPatch = ''
-          # use hyprctl to switch workspaces
-          sed -i 's/zext_workspace_handle_v1_activate(workspace_handle_);/const std::string command = "hyprctl dispatch workspace " + name_;\n\tsystem(command.c_str());/g' src/modules/wlr/workspace_manager.cpp
-        '';
-        mesonFlags = oldAttrs.mesonFlags ++ ["-Dexperimental=true"];
-      });
 
-      xdg-desktop-portal-hyprland = inputs.xdph.packages.${prev.stdenv.hostPlatform.system}.default.override {
-        hyprland-share-picker = inputs.xdph.packages.${prev.stdenv.hostPlatform.system}.hyprland-share-picker.override {inherit hyprland;};
+      wayland-latest = final: prev: {
+        wayland = prev.wayland.overrideAttrs (old: rec {
+          version = "1.22.0";
+          src = final.fetchurl {
+            url = "https://gitlab.freedesktop.org/wayland/wayland/-/releases/${version}/downloads/${old.pname}-${version}.tar.xz";
+            hash = "sha256-FUCvHqaYpHHC2OnSiDMsfg/TYMjx0Sk267fny8JCWEI=";
+          };
+        });
       };
     };
 
     checks = genSystems (system:
-      (lib.filterAttrs (n: _: (lib.hasPrefix "hyprland" n) && !(lib.hasSuffix "debug" n)) self.packages.${system})
-      // {inherit (self.packages.${system}) xdg-desktop-portal-hyprland;});
+      (lib.filterAttrs
+        (n: _: (lib.hasPrefix "hyprland" n) && !(lib.hasSuffix "debug" n))
+        self.packages.${system})
+      // {
+        inherit (self.packages.${system}) xdg-desktop-portal-hyprland;
+      });
 
     packages = genSystems (system:
-      (self.overlays.default null pkgsFor.${system})
+      (self.overlays.default pkgsFor.${system} pkgsFor.${system})
       // {
         default = self.packages.${system}.hyprland;
       });
 
     devShells = genSystems (system: {
-      default = pkgsFor.${system}.mkShell.override {stdenv = pkgsFor.${system}.gcc12Stdenv;} {
-        name = "hyprland-shell";
-        nativeBuildInputs = with pkgsFor.${system}; [
-          cmake
-        ];
-        buildInputs = [
-          self.packages.${system}.wlroots-hyprland
-        ];
-        inputsFrom = [
-          self.packages.${system}.wlroots-hyprland
-          self.packages.${system}.hyprland
-        ];
-      };
+      default =
+        pkgsFor.${system}.mkShell.override {
+          stdenv = pkgsFor.${system}.gcc12Stdenv;
```

Comment:
We can remove this line and others referring to gcc12Stdenv, since it's now the default.